crates/script/src/execute.rs

```diff
@@ -386,13 +386,31 @@ impl ExecutedState {
 }
 
 impl PreSimulationState {
-    pub fn show_json(&self) -> Result<()> {
+    pub async fn show_json(&self) -> Result<()> {
         let result = &self.execution_result;
+        let decoder = &self.execution_artifacts.decoder;
+        let mut traces = result.traces.clone();
+
+        for (_, trace) in &mut traces {
+            decode_trace_arena(trace, decoder).await?;
+        }
+
+        let result_with_decoded_traces = &ScriptResult {
```

Comment:
please add a `#[derive(Clone)]` to `ScriptResult`, clone it and then modify `result.traces` directly

Reply:
Thanks, changes in 5390c6692f96346501316703ea9558073b4502dc